### PR TITLE
Enable testing with `io_uring` back

### PR DESCRIPTION
**Pull request by nickitat**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### CODE REVIEW

**Comment by robot-clickhouse-ci-2** [comment]
*This is an automated comment for commit 52164cda0829b1f67da705fdd0dd72364ee2bcbc with description of existing statuses. It's updated for the latest CI running*

[⏳ Click here](https://s3.amazonaws.com/clickhouse-test-reports/60383/52164cda0829b1f67da705fdd0dd72364ee2bcbc/ci_running.html) to open a full report in a separate page

<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/60383/52164cda0829b1f67da705fdd0dd72364ee2bcbc/ci_running.html">⏳ pending</a></td></tr>
<tbody>
</table>

**Comment by nickitat** [comment]
according to docs, `io_uring_queue_init` return `-errno`, so the error it gives back seems to be `EPERM`

**Comment by Felixoid** [comment]
On one hand, `--privileged` is a solution to many issues.

On the other, let's then try using the more specific `--cap-add=SYS_ADMIN`

**Comment by nickitat** [comment]
it isn't enough: https://s3.amazonaws.com/clickhouse-test-reports/60383/962157d55068eca8801db4ca69e5feeb8723487d/stateful_tests__aarch64_.html

File: src/Storages/StorageFile.cpp [hunk 1]
```diff
@@ -279,22 +276,6 @@ std::unique_ptr<ReadBuffer> selectReadBuffer(
 
         ProfileEvents::increment(ProfileEvents::CreatedReadBufferOrdinary);
     }
-    else if (read_method == LocalFSReadMethod::io_uring && !use_table_fd)
```
Comment (azat) [inline src/Storages/StorageFile.cpp:282]: Why `storage_file_read_method=io_uring` supported had been reverted?

File: tests/clickhouse-test [hunk 2]
```diff
@@ -573,27 +574,6 @@ def get_localzone():
...
-            "select * from file('/dev/null', 'LineAsString')",
-            "--storage_file_read_method",
-            "io_uring",
-        ],
-        stdout=subprocess.DEVNULL,
-        stderr=subprocess.DEVNULL,
-    )
-
-
-def get_local_filesystem_methods():
```
Comment (azat) [inline tests/clickhouse-test:590]: Instead of reverting this, maybe it is better to add a test with **explicit** `io_uring` method requested?

**Comment by azat** [comment]
`SYS_ADMIN` will not help because AFAIR only `--privileged` disables all SELinux policies. So something like this may work `--security-opt label:disable` I guess (though I haven't tried)

File: src/Storages/StorageFile.cpp (hunk 1 above)
Comment (nickitat) [reply src/Storages/StorageFile.cpp]: because I wanted to quickly test the fix

**Review by Felixoid — APPROVED** [review]

**Comment by alexey-milovidov** [comment]
No backport, please. We are not interested in io_uring in releases.

**Pull request merged**